### PR TITLE
Build and release ARM-compatible binary

### PR DESCRIPTION
**Pull request by denismakogon**
Closes: https://github.com/fnproject/fn/issues/431

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,12 +17,18 @@ dep-up:
...
+	GOOS=darwin go build -o fn
+	GOOS=linux go build -o fn
+	go build -o fn
 	./test.sh
 
 release:
 	GOOS=linux go build -o fn_linux
 	GOOS=darwin go build -o fn_mac
 	GOOS=windows go build -o fn.exe
+	GOOS=linux GOARCH=amd64 go build -o fn_linux_arm64
```
Comment (treeder) [inline Makefile]: Isn't this just the normal arch?  Did you mean to use `arm`?

Comment (denismakogon) [reply Makefile]: Yeah. Fixing.

**Comment by denismakogon** [comment]
@treeder can we merge this one? Also solves https://github.com/fnproject/fn/issues/431

**Comment by treeder** [comment]
@denismakogon want to rebase this?

**Comment by denismakogon** [comment]
yeah, would do, thanks

**Comment by denismakogon** [comment]
@treeder can you review this one?

File: Makefile [hunk 2]
```diff
@@ -16,13 +16,20 @@ dep:
 dep-up:
 	dep ensure
 
-test: build
+
+test:
+	GOOS=windows go build -o fn
```
Comment (treeder) [inline Makefile]: Why this change?  Doesn't seem to make much sense.

Comment (denismakogon) [reply Makefile]: because we don't know if we can compile fn until that moment when CI attempts to release fn

Comment (denismakogon) [reply Makefile]: I think that is the valid reason to check if fn compiles for various platforms we consider as "supported"

Comment (treeder) [reply Makefile]: I just mean why build five times, one overwriting the next. Seems like the old way was fine.

Comment (denismakogon) [reply Makefile]: that's the point, each next step overwrites previous and the last step will produce linux-compatible binary that CI can use for further testing.

Comment (treeder) [reply Makefile]: I don't think this is necessary and it doesn't build the correct one for whoever is running this command.

Comment (denismakogon) [reply Makefile]: i extracted that to its own namespace:` test-cross-compile`

**Comment by rikgibson** [comment]
Given no movement here in 7 months I'd suggest closing this.

3...2...1...

**Comment by rdallman** [comment]
think there's a lot to do to support arm https://github.com/fnproject/fn/issues/1128 e.g. `fn start` won't work either even with this. would be cool eventually, just needs some <3

**Comment by rikgibson** [comment]
Think this should just be closed then - it can always be reopened if ever Arm support becomes a priority.

**Comment by Anprotaku** [comment]
Is there any update on this?
